stat/roc_example_test.go

```diff
@@ -6,51 +6,106 @@ package stat_test
 
 import (
 	"fmt"
+	"math"
 
+	"gonum.org/v1/gonum/floats"
 	"gonum.org/v1/gonum/integrate"
 	"gonum.org/v1/gonum/stat"
 )
 
 func ExampleROC_weighted() {
 	y := []float64{0, 3, 5, 6, 7.5, 8}
-	classes := []bool{true, false, true, false, false, false}
+	classes := []bool{false, true, false, true, true, true}
 	weights := []float64{4, 1, 6, 3, 2, 2}
 
-	tpr, fpr := stat.ROC(0, y, classes, weights)
+	tpr, fpr, _ := stat.ROC(nil, y, classes, weights)
 	fmt.Printf("true  positive rate: %v\n", tpr)
 	fmt.Printf("false positive rate: %v\n", fpr)
 
 	// Output:
-	// true  positive rate: [0 0.4 0.4 1 1 1 1]
-	// false positive rate: [0 0 0.125 0.125 0.5 0.75 1]
+	// true  positive rate: [0 0.25 0.5 0.875 0.875 1 1]
+	// false positive rate: [0 0 0 0 0.6 0.6 1]
 }
 
 func ExampleROC_unweighted() {
 	y := []float64{0, 3, 5, 6, 7.5, 8}
-	classes := []bool{true, false, true, false, false, false}
+	classes := []bool{false, true, false, true, true, true}
 
-	tpr, fpr := stat.ROC(0, y, classes, nil)
+	tpr, fpr, _ := stat.ROC(nil, y, classes, nil)
 	fmt.Printf("true  positive rate: %v\n", tpr)
 	fmt.Printf("false positive rate: %v\n", fpr)
 
 	// Output:
-	// true  positive rate: [0 0.5 0.5 1 1 1 1]
-	// false positive rate: [0 0 0.25 0.25 0.5 0.75 1]
+	// true  positive rate: [0 0.25 0.5 0.75 0.75 1 1]
+	// false positive rate: [0 0 0 0 0.5 0.5 1]
+}
```

Comment:
perhaps also add an example where the thresholds are displayed?

perhaps one that matches this one from `sklearn.metrics.roc_curve`?
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_curve.html
```python
>>> import numpy as np
>>> from sklearn import metrics
>>> y = np.array([1, 1, 2, 2])
>>> scores = np.array([0.1, 0.4, 0.35, 0.8])
>>> fpr, tpr, thresholds = metrics.roc_curve(y, scores, pos_label=2)
>>> fpr
array([0. , 0. , 0.5, 0.5, 1. ])
>>> tpr
array([0. , 0.5, 0.5, 1. , 1. ])
>>> thresholds
array([1.8 , 0.8 , 0.4 , 0.35, 0.1 ])
```

Reply:
Done